kolibri/core/assets/src/api-resource.js

```diff
@@ -17,6 +17,18 @@ class Model {
       throw new TypeError('resource must be defined');
     }
 
+    if (!data) {
```

Comment:
Enforced these in the constructor, rather than letting them get through to the set method where they would cause trouble anyway.